### ISSUE TITLE
Diffs in tests under GAP stable-4.9 branch

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/xmodalg#24]
There are diffs in the tests under GAP stable-4.9 branch: see e.g. https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9/jobs/392042904

```
testing: /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/cat1.tst
     896 ms (148 ms GC) and 78.4MB allocated for cat1.tst
testing: /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst:
117
# Input is:
X1 := XModAlgebra( A, B );
# Expected output:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0\
)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 ] )]
# But found:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, 
  (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2 
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 ] )]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst:
124
# Input is:
X2 := XModAlgebra( C, D );
# Expected output:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0\
)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 ] )]
# But found:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, 
  (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2 
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 ] )]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst:
133
# Input is:
Display( mor );
# Expected output:
Morphism of crossed modules :-
: Source = [Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)\
*f1+(Z(2)^0)*f2,
  (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^\
0)*f2 ] )] with generating sets:
  [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*\
f2+(Z(2)^0)*f1*f2 ]
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*f1, (Z(2)^0)*f2 ]
:  Range = [Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)\
*f1+(Z(2)^0)*f2,
  (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^\
0)*f2 ] )] with generating sets:
  [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*\
f2+(Z(2)^0)*f1*f2 ]
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*f1, (Z(2)^0)*f2 ]
: Source Homomorphism maps source generators to:
  [ <zero> of ..., <zero> of ..., <zero> of ... ]
: Range Homomorphism maps range generators to:
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*<identity> of ..., (Z(2)^0)*<identity\
> of ... ]
# But found:
Morphism of crossed modules :- 
: Source = [Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, 
  (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2 
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 
 ] )] with generating sets:
  [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, (Z(2)^0)*f1+(Z(2)^0)*f2, 
  (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ]
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*f1, (Z(2)^0)*f2 ]
:  Range = [Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, 
  (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2 
 ] )->AlgebraWithOne( GF(2), [ (Z(2)^0)*f1, (Z(2)^0)*f2 
 ] )] with generating sets:
  [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, (Z(2)^0)*f1+(Z(2)^0)*f2, 
  (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ]
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*f1, (Z(2)^0)*f2 ]
: Source Homomorphism maps source generators to:
  [ <zero> of ..., <zero> of ..., <zero> of ... ]
: Range Homomorphism maps range generators to:
  [ (Z(2)^0)*<identity> of ..., (Z(2)^0)*<identity> of ..., 
  (Z(2)^0)*<identity> of ... ]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst:
155
# Input is:
X3 := Kernel( mor );
# Expected output:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0\
)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2
 ] )->Algebra( GF(2), [ (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0)*f1*f2, (\
Z(2)^0)*<identity> of ...+(Z(2)^0)*f1
 ] )]
# But found:
[Algebra( GF(2), [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, 
  (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ] )->Algebra( GF(2), 
[ (Z(2)^0)*f1+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0)*f1*f2, 
  (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1 ] )]
########
########> Diff in /home/gap/inst/gap-stable-4.9/pkg/XModAlg-1.16/tst/xmod.tst:
169
# Input is:
Print( MappingGeneratorsImages( theta ), "\n" ); 
# Expected output:
[ [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f2, (Z(2)^0)*f1+(Z(2)^0)*f2, 
      (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ], 
  [ <zero> of ..., <zero> of ..., <zero> of ... ] ]
# But found:
[ [ (Z(2)^0)*<identity> of ...+(Z(2)^0)*f1, (Z(2)^0)*f1+(Z(2)^0)*f2, 
      (Z(2)^0)*f2+(Z(2)^0)*f1*f2 ], 
  [ <zero> of ..., <zero> of ..., <zero> of ... ] ]
########
    1108 ms (176 ms GC) and 85.5MB allocated for xmod.tst
-----------------------------------
total      2004 ms (324 ms GC) and 164MB allocated
#I  Errors detected while testing
```

If the output is mathematically correct, and the diff is due to some randomness involved, perhaps it's possible to make tests more robust by suppressing the output and check some properties of the result instead?

**Comment by cdwensley** [comment]
Is this still a problem?  If so, I will look into it. 
Your email "Release schedule for GAP 4.10" suggests that I could look at the webpage 
https://github.com/gap-system/gap-distribution/  to see the latest log, but that did not seem to work.

**Comment by olexandr-konovalov** [comment]
Still a problem for the xmodalg release from GAP 4.9.2 distro. You need to click on badges in  "require inspection" column to find those logs. Note that xmodalg tests now pass in the master branch, where Semigroups package can not be compiled at the moment - we need to make tests not sensitive to this.

**Comment by cdwensley** [comment]
Logs now found - will attempt to update the tests. 
Just released groupoids-1.57 in which Semigroups is removed from the SuggestedOtherPackages list. 
So neither XMod nor XModAlg will now load Semigroups.

**Comment by cdwensley** [comment]
>   You can now find links to all package tests that are daily run on Travis in 
>   one place here: https://github.com/gap-system/gap-distribution/. If you click
>   on the corresponding badge, the link will lead you to the latest build on Travis. 

Am I misunderstanding what these tests do?  The following test, which ran earlier today: 
          https://travis-ci.org/gap-system/gap-docker-pkg-tests-stable-4.9-staging/jobs/421860092
is not using the commits made yesterday, particularly 70f3393 which changed variable names:
          https://github.com/gap-packages/xmodalg/commit/70f3393898703cced074458a2fb5512de7c5ccfe

**Comment by cdwensley** [comment]
Apologies - looking at the wrong job.  Now only one error!

**Comment by cdwensley** [comment]
Fixed in release 1.17.
